vcl_templates/www.vcl.erb

```diff
@@ -184,9 +184,20 @@ sub vcl_recv {
 
   # FIXME: Prefer a fallback director if we move to Varnish 3
   if (req.restarts > 2) {
-    set req.backend = F_mirror1;
-    set req.http.host = "<%= config.fetch('provider1_mirror_hostname') %>";
-    set req.http.Fastly-Backend-Name = "mirror1";
+    set req.backend = F_mirrorS3;
+    set req.http.host = "<%= config.fetch('s3_mirror_hostname') %>";
+    set req.http.Fastly-Backend-Name = "mirrorS3";
+
+    if (req.url ~ "^(/.*)?/([^#?\s]+)\.(atom|css|csv|docx|gif|html|ico|ics|jpeg|jpg|js|json|ods|odt|pdf|png)([\?#]+.*)?$") {
```

Comment:
This may be wrong but regardless I'm worried about the explicit list of extensions in line 191 which I suspect will be fragile